beacon-chain/core/peerdas/info.go

```diff
@@ -111,7 +111,7 @@ func (custodyInfo *CustodyInfo) ActualGroupCount() uint64 {
 // CustodyGroupCount returns the number of groups we should participate in for custody.
 func (tcgc *targetCustodyGroupCount) Get() uint64 {
 	// If subscribed to all subnets, return the number of custody groups.
-	if flags.Get().SubscribeToAllSubnets {
+	if flags.Get().SubscribeAllColumnSubnets {
```

Comment:
I think `SubscribeToAllSubnets` should be super set of `SubscribeAllColumnSubnets`.

`SubscribeAllColumnSubnets` should subscribe to all data column subnets, and `SubscribeToAllSubnets` should subscribe to all subnets, **including** data column subnet.s

Reply:
My point was mainly to say: `--subscribe-all-subnets` should really subscribe to all subnets, and not act like `--subscribe-to-all-subnets-expected-data-column-subnets`.

But yes I agree we might need more granularity as you proposed in your latest comment.